### PR TITLE
[targetallocator] Add support for targetallocator

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add support for targetallocator as part of the Helm chart.

Target Allocator is a component of the OpenTelemetry Operator.
With this addition, the target allocator is deployed to work in coordination with the daemonset of collectors.
It applies a default configuration applying scrape targets per node.
By default, the Target Allocator looks for all ServiceMonitor and PodMonitor CRDs across all namespaces.
This can be tuned by overriding the service account associated with the Target Allocator.

**Link to Splunk idea:**
https://ideas.splunk.com/ideas/SFXIMMID-I-571

**Testing:**
~Manual testing, helm rendering.~
End to end testing with functional tests.

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 1]
```diff
@@ -243,6 +243,17 @@ receivers:
     listenAddress: 0.0.0.0:9080
   {{- end }}
 
+  {{- if .Values.targetallocator.enabled  }}
+  prometheus/crd:
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:247]: `cdr` is a bit confusing here. Why not `prometheus/ta`?

File: helm-charts/splunk-otel-collector/values.yaml [hunk 2]
```diff
@@ -1242,6 +1242,44 @@ certmanager:
...
+# Target Allocator
+# Notice: Target Allocator related features should be considered to have an alpha
+# maturity level and be experimental. There may be breaking changes or Operator
+# features may be replaced entirely with a better alternative in the future.
+#
+# The Target Allocator is running as a deployment with a replica count of 1.
+# It discovers scraping configurations from ServiceMonitor and PodMonitor CRDs and
+# assigns them to collectors.
+# Related documentation: https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator
+################################################################################
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:1255]: We should also mention that it conflicts with `autodetect.prometheus`. Only one of them should be enabled.

**Comment by dmitryax** [comment]
Thanks, Antoine! Overall looks good. Maybe, later, we could utilize the operator in the subchart to build the TA instead of defining it ourselves. But first, we need to reorganize the `operator.instrumentation` values.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl (hunk 1 above)
Comment (atoulme) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:247]: Sure

File: .chloggen/tasupport.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: new_component
+# The name of the component, or a single word describing the area of concern, (e.g. agent, clusterReceiver, gateway, operator, chart, other)
+component: targetallocator
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add support for targetallocator as part of the Helm chart.
+# One or more tracking issues related to the change
+issues: [689]
```
Comment (jvoravong) [inline .chloggen/tasupport.yaml:8]:
```suggestion
issues: [1360]
```

Comment (atoulme) [reply .chloggen/tasupport.yaml:8]: Heh it's both really

File: helm-charts/splunk-otel-collector/values.yaml (hunk 2 above)
Comment (atoulme) [reply helm-charts/splunk-otel-collector/values.yaml:1247]: It doesn't conflict ; these seem completely independent.

File: helm-charts/splunk-otel-collector/values.yaml [hunk 4]
```diff
@@ -1234,6 +1234,44 @@ certmanager:
...
+
+targetallocator:
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:1249]: Maybe `targetAllocator` to follow helm best practices?

Comment (atoulme) [reply helm-charts/splunk-otel-collector/values.yaml]: renamed

File: helm-charts/splunk-otel-collector/values.yaml (hunk 2 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/values.yaml:1247]: Do you mean that annotated endpoints are never scraped with the target allocator? By conflict, I mean collecting the same metrics twice.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**